core/src/test/java/org/apache/iceberg/rest/responses/TestFetchScanTasksResponseParser.java

```diff
@@ -0,0 +1,180 @@
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid fetch scan tasks response: null or empty");
+  }
+
+  @Test
+  public void roundTripSerdeWithEmptyObject() {
+    assertThatThrownBy(
+            () -> FetchScanTasksResponseParser.toJson(FetchScanTasksResponse.builder().build()))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: planTasks and fileScanTask cannot both be null");
+
+    String emptyJson = "{ }";
+    assertThatThrownBy(() -> FetchScanTasksResponseParser.fromJson(emptyJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid fetch scan tasks response: null or empty");
+  }
+
+  @Test
+  public void missingRequiredField() {
+    String missingRequiredFieldJson = "{\"x\": \"val\"}";
+    assertThatThrownBy(() -> FetchScanTasksResponseParser.fromJson(missingRequiredFieldJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid response: planTasks and fileScanTask cannot both be null");
+  }
+
+  @Test
+  public void roundTripSerdeWithPlanTasks() {
+    String expectedJson = "{\"plan-tasks\":[\"task1\",\"task2\"]}";
+    String json =
+        FetchScanTasksResponseParser.toJson(
+            FetchScanTasksResponse.builder().withPlanTasks(List.of("task1", "task2")).build());
+    assertThat(json).isEqualTo(expectedJson);
+
+    FetchScanTasksResponse fromResponse = FetchScanTasksResponseParser.fromJson(json);
+
+    // can't do an equality comparison on PlanTableScanRequest because we don't implement
+    // equals/hashcode
+    assertThat(FetchScanTasksResponseParser.toJson(fromResponse, false)).isEqualTo(expectedJson);
+  }
+
+  @Test
+  public void roundTripSerdeWithDeleteFilesNoFileScanTasksPresent() {
+    assertThatThrownBy(
+            () ->
+                FetchScanTasksResponse.builder()
+                    .withPlanTasks(List.of("task1", "task2"))
+                    .withDeleteFiles(List.of(FILE_A_DELETES))
+                    .build())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Invalid response: deleteFiles should only be returned with fileScanTasks that reference them");
+
+    String invalidJson =
+        "{\"plan-tasks\":[\"task1\",\"task2\"],"
+            + "\"delete-files\":[{\"spec-id\":0,\"content\":\"POSITION_DELETES\","
+            + "\"file-path\":\"/path/to/data-a-deletes.parquet\",\"file-format\":\"PARQUET\","
+            + "\"partition\":{\"1000\":0},\"file-size-in-bytes\":10,\"record-count\":1}]"
+            + "}";
+
+    assertThatThrownBy(() -> FetchScanTasksResponseParser.fromJson(invalidJson))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Invalid response: deleteFiles should only be returned with fileScanTasks that reference them");
+  }
+
+  @Test
+  public void roundTripSerdeWithFileScanTasks() {
```

Comment:
I know `roundTrip` prefix was probably copied from existing tests, but this protocol doesn't really round trip like other JSON messages due to the bound/unbound specs involved. Maybe just call this `serdeWithFileScanTasks`

Reply:
Yea this is a good point will remove the `roundTrip` naming from all the response parser tests